lib/CL/pocl_llvm_api.cc

```diff
@@ -1529,8 +1556,25 @@ static PassManager& kernel_compiler_passes
 	  passes.push_back("automatic-locals");
   if (SPMDDevice)
     passes.push_back("flatten-inline-all");
-  else
-    passes.push_back("flatten-globals");
+  else {
+    /* LLVM is unfortunately still not intelligent enough to flatten.
```

Comment:
By flatten you might mean 'inline' here. So you mean LLVM is not intelligently enough choosing what to inline if we do not perform the inline-everything (flatten) pass?